example/src/credo/CredoScreen.tsx

```diff
@@ -0,0 +1,66 @@
+import { Central, CentralProvider, Peripheral, PeripheralProvider } from '@animo-id/react-native-ble-didcomm'
+import AgentProvider from '@credo-ts/react-hooks'
+import CredentialProvider from '@credo-ts/react-hooks/build/CredentialProvider'
+import ProofProvider from '@credo-ts/react-hooks/build/ProofProvider'
+import * as React from 'react'
+import { type ReactElement, useEffect, useState } from 'react'
+import { Button, Text } from 'react-native'
+import { Spacer } from '../Spacer'
+import { Prover } from './Prover'
+import { Verifier } from './Verifier'
+import { type AppAgent, setupAgent } from './agent'
+
+const uuid = 'd5477fcb-6e8b-4091-ad39-a1e30386ef76'
+
+export const CredoScreen = () => {
+  const [role, setRole] = useState<'prover' | 'verifier'>()
+  const [agent, setAgent] = useState<AppAgent>()
+
+  useEffect(() => {
+    setupAgent().then(setAgent)
+  }, [])
```

Comment:
I've had issues with this in past, where setupAgent was called multiple times. I think we should handle it with state, or we can make it a `useSetupAgent` hook, which handles it with state internally ( and returns the agent afterwards ).

I know it's an example, but good to make the example follow best practices